File: CS/DXSample/ViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using System;

namespace DXSample {
    public class ViewModel : ViewModelBase {
        private Customer Customer { get; set; }
        private bool IsReviewMode { get; set; }
        public MetadataExtendedSource Source {
            get { return GetProperty(() => Source); }
            set { SetProperty(() => Source, value); }
        }

        public ViewModel() {
            Customer = new Customer() {
                FirstName = "Anita",
                LastName = "Benson",
                BirthDate = new DateTime(1985, 3, 18),
                Phone = "(713) 863 813",
                Email = "@gmail.com",
            };
            Source = new MetadataExtendedSource(Customer, MetadataLocator.Create().AddMetadata(typeof(Customer), typeof(ReviewMetadata)));
            IsReviewMode = true;
        }
        [Command]
        public void ChangeSource() {
            if (IsReviewMode) {
                Source = new MetadataExtendedSource(Customer, MetadataLocator.Create().AddMetadata(typeof(Customer), typeof(Metadata)));
                IsReviewMode = false;
            } else {
                Source = new MetadataExtendedSource(Customer, MetadataLocator.Create().AddMetadata(typeof(Customer), typeof(ReviewMetadata)));
                IsReviewMode = true;
            }
        }
    }

    public class Customer {
        public int ID { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string FullName { get { return string.Format("{0} {1}", FirstName, LastName); } }
        public string Phone { get; set; }
        public string Email { get; set; }
        public DateTime BirthDate { get; set; }
    }
}

File: VB/DXSample/ViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports System

Namespace DXSample

    Public Class ViewModel
        Inherits ViewModelBase

        Private Property Customer As Customer

        Private Property IsReviewMode As Boolean

        Public Property Source As MetadataExtendedSource
            Get
                Return GetProperty(Function() Me.Source)
            End Get

            Set(ByVal value As MetadataExtendedSource)
                SetProperty(Function() Source, value)
            End Set
        End Property

        Public Sub New()
            Customer = New Customer() With {.FirstName = "Anita", .LastName = "Benson", .BirthDate = New DateTime(1985, 3, 18), .Phone = "(713) 863 813", .Email = "@gmail.com"}
            Source = New MetadataExtendedSource(Customer, MetadataLocator.Create().AddMetadata(GetType(Customer), GetType(ReviewMetadata)))
            IsReviewMode = True
        End Sub

        <Command>
        Public Sub ChangeSource()
            If IsReviewMode Then
                Source = New MetadataExtendedSource(Customer, MetadataLocator.Create().AddMetadata(GetType(Customer), GetType(Metadata)))
                IsReviewMode = False
            Else
                Source = New MetadataExtendedSource(Customer, MetadataLocator.Create().AddMetadata(GetType(Customer), GetType(ReviewMetadata)))
                IsReviewMode = True
            End If
        End Sub
    End Class

    Public Class Customer

        Public Property ID As Integer

        Public Property FirstName As String

        Public Property LastName As String

        Public ReadOnly Property FullName As String
            Get
                Return String.Format("{0} {1}", FirstName, LastName)
            End Get
        End Property

        Public Property Phone As String

        Public Property Email As String

        Public Property BirthDate As Date
    End Class
End Namespace
